src/web/handlers/PanGestureHandler.ts

```diff
@@ -351,6 +361,70 @@ export default class PanGestureHandler extends GestureHandler {
     }
   }
 
+  private scheduleWheelEnd(event: AdaptedEvent) {
+    this.endWheelTimeout = setTimeout(() => {
+      if (this.receivedWheelEvent) {
+        return;
+      }
+
+      this.end();
+      this.tracker.removeFromTracker(event.pointerId);
+      this.currentState = State.UNDETERMINED;
+
+      this.wheelDevice = WheelDevice.UNDETERMINED;
+    }, 30);
+
+    this.receivedWheelEvent = false;
+  }
+
+  protected onWheel(event: AdaptedEvent): void {
+    if (
+      this.wheelDevice === WheelDevice.MOUSE ||
+      !this.enableTrackpadTwoFingerGesture
+    ) {
+      return;
+    }
+
+    this.receivedWheelEvent = true;
+    clearTimeout(this.endWheelTimeout);
+
+    if (this.currentState === State.UNDETERMINED) {
+      this.wheelDevice =
+        event.wheelDeltaY! % 120 !== 0
+          ? WheelDevice.TOUCHPAD
+          : WheelDevice.MOUSE;
+
+      if (this.wheelDevice === WheelDevice.MOUSE) {
+        this.scheduleWheelEnd(event);
```

Comment:
1. Do we need to schedule in this case instead of ending outright?
2. Do we need to end here at all? `begin` is called after this condition, so we shouldn't even get there with mouse, no?

Reply:
If we don't schedule cleanup, wheelDevice won't be set to `UNDETERMINED`. I'm not sure if we want to set it here though. However, we can call `end` only if handler was `active`.

Reply:
Also changed in [a640c0c](https://github.com/software-mansion/react-native-gesture-handler/pull/3163/commits/a640c0c7700d96d66b95ecda06b1bf8c7bd730c3)